src/components/workspace/diagrams/common/navigation-sidebar.tsx

```diff
@@ -0,0 +1,132 @@
+    if (isAbsolutePositioned && shouldBeCollapsed) {
+        return 'none';
+    }
+    return `1px solid ${theme.palette.divider}`;
+};
+
+const styles = {
+    sidebar: (theme: Theme, shouldBeCollapsed: boolean, isAbsolutePositioned: boolean) => ({
+        width: shouldBeCollapsed ? COLLAPSED_WIDTH : EXPANDED_WIDTH,
+        borderRight: getBorderRight(theme, shouldBeCollapsed, isAbsolutePositioned),
+        borderLeft: isAbsolutePositioned ? 'none' : `1px solid ${theme.palette.divider}`,
+        backgroundColor: getBackgroundColor(theme, shouldBeCollapsed),
+        display: 'flex',
+        flexDirection: 'column' as const,
+        flexShrink: 0,
+        overflowX: 'hidden' as const,
+        ...(shouldBeCollapsed &&
+            isAbsolutePositioned && {
+                position: 'absolute' as const,
+                left: 0,
+                top: 0,
+                bottom: 0,
+                zIndex: 100,
+                pointerEvents: 'none' as const,
+            }),
+    }),
+    header: (theme: Theme, hasHistory: boolean) => ({
+        display: 'flex',
+        alignItems: 'center',
+        p: 1,
+        cursor: hasHistory ? 'pointer' : 'default',
+        pointerEvents: 'auto' as const,
+        '&:hover': {
+            backgroundColor: hasHistory ? theme.palette.action.hover : 'transparent',
+        },
+    }),
+    icon: (theme: Theme, hasHistory: boolean) => ({
+        color: hasHistory ? theme.palette.text.primary : theme.palette.text.disabled,
+    }),
+    list: {
+        flex: 1,
+        overflow: 'auto' as const,
+        py: 0,
+    },
+};
+
+export const NavigationSidebar = memo(function NavigationSidebar({
+    navigationHistory,
+    shouldBeCollapsed,
+    hasHistory,
+    isDisabled,
+    isAbsolutePositioned = false,
+    isItemSelected,
+    onToggleCollapse,
+    onNavigate,
+}: NavigationSidebarProps) {
+    const theme = useTheme();
+    const intl = useIntl();
+
+    return (
+        <Box sx={styles.sidebar(theme, shouldBeCollapsed, isAbsolutePositioned)}>
+            {/* Header */}
+            <Box onClick={hasHistory ? onToggleCollapse : undefined} sx={styles.header(theme, hasHistory)}>
+                <HistoryIcon sx={styles.icon(theme, hasHistory)} />
+                {!shouldBeCollapsed && (
+                    <Typography variant="caption" sx={{ ml: 1, fontWeight: 'medium' }}>
+                        {intl.formatMessage({ id: 'history' })}
+                    </Typography>
+                )}
+            </Box>
+
+            {/* List */}
+            {!shouldBeCollapsed && (
+                <List dense sx={styles.list}>
+                    {navigationHistory.map((voltageLevelId, index) => (
+                        <ListItemButton
+                            key={`${voltageLevelId}-${index}`}
+                            selected={isItemSelected ? isItemSelected(voltageLevelId) : false}
+                            onClick={() => !isDisabled && onNavigate(voltageLevelId)}
```

Comment:
```suggestion
                            onClick={() => onNavigate(voltageLevelId)}
```
disabled is already doing this check I think